### PR TITLE
dotCMS/core#15673 Upgrade to Angular 7 & dotCMS/core#15455 Make listing-data-table.component navegable with keyboard

### PR DESCRIPTION
**Pull request by alfredo-dotcms**

### CODE REVIEW

File: src/app/view/components/_common/pattern-library/pattern-library.component.html [hunk 1]
```diff
@@ -15,6 +15,29 @@ <h1>Hello World</h1>
     <h1>Hello World</h1>
 </dot-dialog>
 
+<h3 class="first">Single Row Selection</h3>
+<p-table [columns]="cols2" [value]="cars" selectionMode="single" [(selection)]="selectedCar1" dataKey="vin">
```
Comment (fmontes) [inline src/app/view/components/_common/pattern-library/pattern-library.component.html]: Shouldn't we add here OUR listing table component?

File: src/styles/primeng-theme/components/_datatable.scss [hunk 2]
```diff
@@ -1,7 +1,12 @@
...
+@import "variables";
+
+.ui-datatable,
+.ui-table {
     .ui-datatable-thead > tr > th,
+    .ui-table-thead > tr > th,
     .ui-datatable-tfoot > tr > td,
-    .ui-datatable-data > tr > td {
+    .ui-datatable-data > tr > td,
+    .ui-table-tbody > tr > td {
```
Comment (fmontes) [inline src/styles/primeng-theme/components/_datatable.scss]: We need an `ui-table` file.

File: src/app/portlets/dot-edit-page/content/services/dot-container-contentlet.service.spec.ts [hunk 3]
```diff
@@ -42,11 +42,9 @@ describe('DotContainerContentletService', () => {
                 type: 'content_type'
             };
 
-            let response;
-
             dotContainerContentletService
                 .getContentletToContainer(pageContainer, pageContent)
-                .subscribe((resp) => (response = resp));
+                .subscribe((resp) => resp);
```
[the remark this answers is not in the document]
Comment (alfredo-dotcms) [reply src/app/portlets/dot-edit-page/content/services/dot-container-contentlet.service.spec.ts]: ng7 throws an error cuz `response` it's not been used, the change was made here: https://github.com/dotCMS/core-web/pull/889

File: src/app/view/components/_common/pattern-library/pattern-library.component.html (hunk 1 above)
Comment (alfredo-dotcms) [reply src/app/view/components/_common/pattern-library/pattern-library.component.html]: I preferred to add the component alone, so anyone can use it without needing the `ListingDataTableComponent`

File: package.json [hunk 4]
```diff
@@ -18,61 +18,61 @@
     },
     "private": false,
     "dependencies": {
-        "@angular/platform-server": "^6.1.9",
+        "@angular/cdk": "^7.2.0",
```
Comment (fmontes) [inline package.json]: Why do you add the `cdk`?

File: package.json [hunk 5]
```diff
@@ -18,61 +18,61 @@
...
         "angular2-tinymce": "^2.1.2",
-        "core-js": "^2.5.3",
+        "core-js": "^2.5.4",
         "font-awesome": "^4.7.0",
         "lodash": "^4.17.11",
         "md5": "^2.2.1",
         "moment": "^2.19.4",
         "ng2-ace-editor": "^0.3.8",
         "ng2-dragula": "^2.1.0",
         "primeicons": "^1.0.0-beta.10",
```
Comment (fmontes) [inline package.json]: There is a final 1.0.0 version of primeicons.

File: src/app/view/components/listing-data-table/listing-data-table.component.html [hunk 6]
```diff
@@ -1,47 +1,53 @@
 <dot-action-header [options]="actionHeaderOptions">
```
Comment (fmontes) [inline src/app/view/components/listing-data-table/listing-data-table.component.html]: This component is not working as expected, if you use it with 10 items per pege here: https://github.com/dotCMS/core-web/blob/issue-15455-Make-listing-data-table-component-navegable-with-keyboard/src/app/portlets/content-types/main/content-types.component.html#L7 is not showing the pagination correctly.

File: package.json (hunk 4 above)
Comment (alfredo-dotcms) [reply package.json]: it's needed to compile `ng build dotcms-js`

File: package.json (hunk 5 above)
Comment (alfredo-dotcms) [reply package.json]: thanks to the `^` even when declared as `1.0.0-beta` it'll get the most recent major version 
https://stackoverflow.com/a/22345808

*Anyway I updated it

File: package.json (hunk 4 above)
Comment (fmontes) [reply package.json]: Are you 100% about this? Because we didn't have this on 6 and were able to compile `dotcms-js`

File: package.json [hunk 7]
```diff
@@ -18,61 +18,56 @@
...
-        "@types/jasmine": "~2.5.53",
-        "@types/jasminewd2": "~2.0.2",
-        "@types/node": "^6.0.85",
-        "@types/uglify-js": "^2.6.28",
-        "codelyzer": "~4.2.1",
-        "jasmine-core": "~2.6.2",
-        "jasmine-spec-reporter": "~4.1.0",
-        "karma": "~1.7.0",
-        "karma-chrome-launcher": "~2.1.1",
-        "karma-cli": "~1.0.1",
```
Comment (fmontes) [inline package.json]: Removed, not used.

File: package.json [hunk 8]
```diff
@@ -18,61 +18,56 @@
...
         "ng2-ace-editor": "^0.3.8",
         "ng2-dragula": "^2.1.0",
-        "primeicons": "^1.0.0-beta.10",
-        "primeng": "^6.1.4",
-        "zone.js": "^0.8.26"
+        "primeicons": "^1.0.0",
+        "primeng": "^7.0.3"
     },
     "devDependencies": {
-        "@angular-devkit/build-angular": "~0.7.0",
```
Comment (fmontes) [inline package.json]: All this @angular/xxx goes into dependencies

File: package.json [hunk 9]
```diff
@@ -18,61 +18,56 @@
     },
     "private": false,
     "dependencies": {
-        "@angular/platform-server": "^6.1.9",
```
Comment (fmontes) [inline package.json]: Removed, not used.

File: package.json [hunk 10]
```diff
@@ -18,61 +18,56 @@
...
+        "@types/jasminewd2": "~2.0.3",
+        "codelyzer": "~4.5.0",
+        "jasmine-core": "~2.99.1",
+        "jasmine-spec-reporter": "~4.2.1",
+        "karma": "~3.1.1",
+        "karma-chrome-launcher": "~2.2.0",
+        "karma-coverage-istanbul-reporter": "^2.0.1",
+        "karma-jasmine": "~1.1.2",
         "karma-jasmine-html-reporter": "^0.2.2",
-        "ng-cli": "^0.7.0",
```
Comment (fmontes) [inline package.json]: Removed, not used.

File: package.json [hunk 11]
```diff
@@ -18,61 +18,56 @@
...
-        "ng-packagr": "^3.0.0",
-        "node-sass": "^4.9.0",
```
Comment (fmontes) [inline package.json]: Removed, not used.

File: package.json [hunk 12]
```diff
@@ -18,61 +18,56 @@
...
-        "@angular/http": "^6.1.9",
-        "@angular/language-service": "^6.1.9",
-        "@angular/platform-browser": "^6.1.9",
-        "@angular/platform-browser-dynamic": "^6.1.9",
-        "@angular/router": "^6.1.9",
-        "@types/googlemaps": "^3.30.1",
-        "@types/jasmine": "~2.5.53",
-        "@types/jasminewd2": "~2.0.2",
-        "@types/node": "^6.0.85",
-        "@types/uglify-js": "^2.6.28",
```
Comment (fmontes) [inline package.json]: Removed, not used.

File: package.json [hunk 13]
```diff
@@ -18,61 +18,56 @@
...
         "ng2-dragula": "^2.1.0",
-        "primeicons": "^1.0.0-beta.10",
-        "primeng": "^6.1.4",
-        "zone.js": "^0.8.26"
+        "primeicons": "^1.0.0",
+        "primeng": "^7.0.3"
     },
     "devDependencies": {
-        "@angular-devkit/build-angular": "~0.7.0",
-        "@angular-devkit/build-ng-packagr": "~0.7.0",
```
Comment (fmontes) [inline package.json]:
```
"@angular-devkit/build-ng-packagr": "~0.11.4",
"ng-packagr": "^4.4.5",
"tsickle": ">=0.31.0"
```

This are used by Angular when you create a library inside the app.

File: projects/dotcms-js/src/lib/core/util/protocol.ts [hunk 14]
```diff
@@ -64,7 +64,7 @@ export abstract class Protocol {
 
     protected reconnect(): void {
         this.destroy();
-        setTimeout(this.start(), 0);
+        setTimeout(() => { this.start(); }, 0);
```
Comment (freddyucv) [inline projects/dotcms-js/src/lib/core/util/protocol.ts]: I think that this timeout is not really needed

**Pull request merged**